### ISSUE TITLE
Crash/error/freeze if undo is pressed when selection is on page boundary 

### ISSUE THREAD

**Issue by Technius** [xournalpp/xournalpp#1232]
**Affects versions :**
 - OS: Linux
 - Version of Xournal++: master as of 99b7e18, most likely in earlier versions too

**Describe the bug**
Pressing an undo button when a selection is between two pages will cause an error (not user visible) and do nothing else. It may also cause the entire application to freeze (which I have not been able to reproduce consistently yet). If there is no freeze, a crash will occur when changing documents. The resulting emergency save will have no pages when loaded (no emergency save is produced if the application freezes).

**To Reproduce**
1. Create a blank document and add a second page.
2. Create some strokes on the first page.
3. Select the stroke and move it on to the page boundary so that it is mostly on the second page but still on the first page.
4. Trigger an undo (e.g. by pressing <kbd>Ctrl</kbd><kbd>z</kbd>)
5. Observe the error in the console (or a freeze).
6. If there is no freeze, create a new document.
7. Observe a crash.

**Expected behavior**
The undo should occur successfully without any errors.

**Screenshots of Problem**
![xournal_crash](https://user-images.githubusercontent.com/1066652/59314159-dd5f2b80-8c68-11e9-9360-b344ca34fcce.gif)

**Additional context**
[crashlog.log](https://github.com/xournalpp/xournalpp/files/3278979/crashlog.log)

**Comment by LittleHuba** [comment]
I can explain some background on this problem. It is mainly caused by the active page system.
Objects on pages are not allowed to be placed with negative coordinates for whatever reason. Thus moving the stroke selects the second page as active page and the stroke returns to the original position as its new coordinates would be negative.
If you try this with the mouse staying on page one everything should work.

So this has two fixes:
1) Allow negative coordinates (probably the best as this would be what one expects)
2) Fix that active page system to still assign the correct page

**Comment by lehmanju** [comment]
@Technius Is this still reproducable? I can't seem to reproduce the bug. In your screenshot it also seems the page selection is not quite working, in the sidebar the first page is selected whereas the second page is selected in the main view.

**Comment by Technius** [comment]
Yes, I still get the error in the console as well as a segfault in current master.

**Comment by rolandlo** [comment]
I can't reproduce this bug any more. It must have be fixed.

**Comment by Technius** [comment]
Assuming that there are rare cases where it is possible to place a selection "out" of a page, this bug can still occur (as evidenced by #2889, which isn't really a "rare" case but demonstrates the possibility).

**Comment by tattsan** [comment]
This occurs if we move the selected object to another page.
If we undo then, the object in the original page won't be removed.
```
(xournalpp:151913): Gtk-WARNING **: 21:47:58.419: Failed to measure available space: The specified location is not supported
** (xournalpp:151913): WARNING **: 21:48:03.032: Could not remove element from layer, it's not on the layer!
```
In my case, there's no crash or freeze.

**Comment by rolandlo** [comment]
> Not limited to this case. This occurs if we move the selected object to another page.
> If we undo then, the object in the original page won't be removed.
> 
> ```
> (xournalpp:151913): Gtk-WARNING **: 21:47:58.419: Failed to measure available space: The specified location is not supported
> ** (xournalpp:151913): WARNING **: 21:48:03.032: Could not remove element from layer, it's not on the layer!
> ```

I can reproduce this. The problem only appears if the object is selected when undoing the move. This looks like a bug in the `MoveUndoAction` logic.
